packages/site/src/lib/components/Agenda.svelte

```diff
@@ -20,19 +30,60 @@
     viewBox: "0 0 48 48",
     style: "width: calc(6 * var(--space)); height: calc(6 * var(--space))",
   };
+
+  const getFilteredEvents = (): Event[] => {
+    if((searchValue ?? '') === '') {
+      return events
+    }
+    
+    return events.filter((ev) => ev.name.toLowerCase().includes(searchValue.toLowerCase()))
+  };
+
+  const getCurrentPageEvents = (): Event[] => {
+    const start = (pageNumber - 1) * maxPerPage;
+    const end = start + maxPerPage
+    return filteredEvents.slice(start, end);
+  };
+
+  const updatePagedEvents = (): void => {
+    currentPageEvents = getCurrentPageEvents();
+    numPages = Math.ceil(filteredEvents.length / maxPerPage);
+  };
+
+  const goBack = (): void => {
+    pageNumber -= 1;
+    updatePagedEvents();
+  };
+
+  const advance = (): void => {
+    pageNumber += 1;
+    updatePagedEvents();
+  };
+
+  const updateEvents = (): void => {
+    filteredEvents = getFilteredEvents()
+    updatePagedEvents();
+  };
+
+  updatePagedEvents();
```

Comment:
```suggestion
  $: canAdvance: number = pageNumber < numPages
  $: canGoBack: number = pageNumber > 1

  const goBack = (): void => {
    if (canGoBack) {
      pageNumber -= 1;
    }
  };

  const advance = (): void => {
    if (canAdvance) {
      pageNumber += 1;
    }
  };
```

Probably should check this here, plus this gives these concepts a name.